snap/snaptest/snaptest.go

```diff
@@ -154,6 +154,18 @@ func MockInvalidInfo(c *check.C, yamlText string, sideInfo *snap.SideInfo) *snap
 	return snapInfo
 }
 
+// MockSnapWithFiles does the same as MockSnap, but also populates the snap
+// directory with given content
+//
+// The caller is responsible for mocking root directory with dirs.SetRootDir()
+//and for altering the overlord state if required.
+func MockSnapWithFiles(c *check.C, yamlText string, si *snap.SideInfo, files [][]string) *snap.Info {
```

Comment:
Note this will create a snap that will likely fail snap checks (e.g. due to missing exec bits on apps). That's probably fine for many tests that avoid snap checks though.

Reply:
Yes, that's intentional. Once, there's a need we can switch the internal implementation to use osutil.EnsureDirState()